src/rules.ts

```diff
@@ -0,0 +1,382 @@
+import {
+  Question,
+  TotalInformationState,
+  InformationState,
+  Move,
+  DomainRelation,
+} from "./types";
+
+type Rules = {
+  [index: string]: (context: TotalInformationState) => {
+    preconditions: boolean;
+    effects: InformationState;
+  };
+};
+
+function relevant(x: DomainRelation): boolean {
+  return ["relevant", "resolves"].includes(x.type);
+}
+function resolves(x: DomainRelation): boolean {
+  return "resolves" === x.type;
+}
+
+const plans = {
+  create_appointment: [
+    {
+      type: "findout",
+      content: {
+        type: "wh_question",
+        predicate: "meeting_person",
+      },
+    },
+  ],
+};
+
+export const rules: Rules = {
+  clear_agenda: ({ is }) => {
+    const newIS = {
+      ...is,
+      private: { ...is.private, agenda: [] },
+    };
+    console.debug(`[ISU clear_agenda]`, newIS);
+    return {
+      preconditions: true,
+      effects: newIS,
+    };
+  },
+
+  /**
+   * Grounding
+   */
+  get_latest_move: (context) => {
+    const newIS = {
+      ...context.is,
+      shared: {
+        ...context.is.shared,
+        lu: {
+          move: context.latest_move!,
+          speaker: context.latest_speaker!,
+        },
+      },
+    };
+    console.debug(`[ISU get_latest_move]`, newIS);
+    return {
+      preconditions: true,
+      effects: newIS,
+    };
+  },
+
+  /**
+   * Integrate
+   */
+  /** rule 5.1 */
+  integrate_usr_request: ({ is }) => {
+    if (
+      is.shared.lu!.speaker === "usr" &&
+      is.shared.lu!.move.type === "request"
```

Comment:
Should be added to `types.ts` (`interface Move` definition). Perhaps, we should also make distinction between moves and actions.